### DISCUSSION TITLE
Is there a way to catch non-mocked responses?

### DISCUSSION THREAD

**Discussion by hornta** [sinonjs/nise#233] [General]
I've setup my test suite to mock every response my application performs to our REST API but I feel like it's common to forget to mock some API:s so I would want a way for our tests to fail when a non-mocked API is called upon. Is something like that possible with nise/sinon?

**Comment by fatso83** [comment]
Not that I can think of immediately, but this is kind of like a catch-all if no other. Would be interesting to have this feature. Can you have a little browse of the (not very big) source code and see if the match code can be made to do this? Maybe it can accept a function or something.

**Comment by hornta** [comment]
The fakeserver can take a catch-all response but it doesn't help me being able to throw when that response is matched. I tend to use https://mswjs.io/ in other projects so inspiration can be taken from there https://mswjs.io/docs/recipes/debugging-uncaught-requests

Maybe the fakeserver can emit an event like "onUnhandledRequest"?

**Reply by fatso83** [reply]
The idea is great, but I am not using this library actively in my current work, so if someone wants this I'd look at what it takes to implement it.

[a comment by an automated account is not shown]
